File: examples/sdk/typescript/readiness.ts
/**
 * Evaluate readiness via the Spanda TypeScript SDK.
 *
 *   npx tsx examples/sdk/typescript/readiness.ts
 */
import { SpandaClient } from "../../../sdk/typescript/src/index.js";

async function main(): Promise<void> {
  const client = SpandaClient.local();
  const report = await client.readiness("examples/robotics/rover.sd");
  console.log("Readiness score:", report.score ?? report.raw);
}

main().catch(console.error);

File: examples/sdk/python/readiness.py
#!/usr/bin/env python3
"""Evaluate program readiness via the Spanda Python SDK."""

from spanda import SpandaClient


def main() -> None:
    client = SpandaClient.local()
    report = client.readiness("examples/robotics/rover.sd")
    score = report.get("report", {}).get("score", {})
    total = score.get("total") if isinstance(score, dict) else score
    print(f"Readiness score: {total}")


if __name__ == "__main__":
    main()
